### DISCUSSION TITLE
Apollo @client local fields [not related to directives]

### DISCUSSION THREAD

**Discussion by glebdv** [graphql-hive/graphql-eslint#1515] [Q&A]
Hello!

First off, I want to say that you all have created a phenomenal library. I've been setting it up in the past day (we have a fairly legacy setup, so some mental gymnastics were involved), and I've been enjoying the results a lot.

To get to the issue/question at hand^^

Apollo allows the creation of "local" fields which we mostly use in this fashion:

>     query getUserLabels($language: String!) {
>       users {
>          label {
>             default
>             en
>             fr
>          }
>          translatedLabel (language: $language) @client
>       }
>     }
(default, en, and fr are all "string"s)

where right after the data is received by the frontend and before it's fed into a component, `translatedLabel` goes through the label object, and ouputs the string that matches the current language. Basically, the same thing as doing it with a function in the frontend after the data is received, but this way it returns the data in the query body right away.

The issue: naturally, these fields like `translatedLabel` are not present in the schema because they are added client side, and they fail the `@graphql-eslint/fields-on-correct-type`. I don't think I can add them manually - our schema regenerates with a script, triggered manually (yup), and is usually done after server-side schema changes. I thought about injecting the schema with these values, but im dropping that idea for now because it feels a little barbaric (every time we add this to a new type, we would have to add it in the script as well, and it seems like a pain).

I thought about doing what was mentionned here: 
https://github.com/B2o5T/graphql-eslint/discussions/1063#discussioncomment-2721160
but I don't think it is an issue of directives? (I already have 
`"@graphql-eslint/known-directives": ["error", { ignoreClientDirectives: ["client"] }]` 
to ignore the @client) and I'm not sure I understand how to extend for example this `User` type with a `translatedLabel: String` field manually in the parserOptions/schema 

My last resort is making a custom rule to check if it has a `@client` directive and try to ignore that query field? 

Is there something I overlooked/missed? If anyone had tips for this, it would be very appreciated!

**Comment by glebdv** [comment]
Well, I just ended up making a custom eslint rule to omit the local fields that we have created.
Feels a tiiny bit off, because I would have thought that there would be a way to mute these errors on the 
`fields-on-correct-type` rule, kind of the same way that `known-directive` rule allows to ignore a client directive
(ex: `"@graphql-eslint/known-directives": ["error", { "ignoreClientDirectives": ["client"] }]` but hey. If I get around it,
I might create a proposal for this

**Comment by dimaMachina** [comment]
Thanks! I described how to extend your schema here https://github.com/B2o5T/graphql-eslint/discussions/1063#discussioncomment-2721295
so following should works
```graphql
extend type User {
  someClientField: String!
}
```

**Reply by glebdv** [reply]
I tried and it doesn't pick up the `extend`, but that's not your fault. It's probably the way that our setup is 
(we have `relay-classic` (aka relay 1.0) that im replacing, but it keeps a lot of packages behind because it's very fragile. Our Eslint is at 6.8.0 (release at 8.36.0), and many more inconsistancies like that which make a regular setup near impossible and would make debugging harder, so sadly its not worth for me to try to reproduce this monstrocity in a sandbox ^_^ 
I appreciate you giving me an example though! I'll use it when ill be able to update all the dependencies without breaking the project. Spasibo and stay awesome! ))
